src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
 
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
+
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        if (cache.containsKey(userId)) {
+            return CompletableFuture.completedFuture(cache.get(userId));
+        }
+        return CompletableFuture.supplyAsync(() -> {
+            try {
+                Thread.sleep(500);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            UserData userData = new UserData(userId, "Details for user: " + userId);
+            cache.put(userId, userData);
+            return userData;
+        }, executor);
```

Comment:
1) Please move logic for item creation into separate method
2) check computeIfAbsent map method, seems it may make your code more simple